src/t_hash.c

```diff
@@ -230,12 +230,12 @@ int hashTypeSet(robj *o, sds field, sds value, int flags) {
             /* Push new field/value pair onto the tail of the listpack */
             zl = lpAppend(zl, (unsigned char*)field, sdslen(field));
             zl = lpAppend(zl, (unsigned char*)value, sdslen(value));
+
+            /* Check if the listpack needs to be converted to a hash table */
+            if (hashTypeLength(o) > server.hash_max_listpack_entries)
+                hashTypeConvert(o, OBJ_ENCODING_HT);
         }
         o->ptr = zl;
```

Comment:
if listpack is converted to hash above, `zl` has been freed in `hashTypeConvert()`.

Reply:
move `o->ptr` to a statment behind `lpReplace`